assets/js/googlesitekit/components-gm3/Dialog.js

```diff
@@ -16,6 +16,48 @@
  * limitations under the License.
  */
 
-export default function Dialog() {
-	return null;
+import { MdDialog } from '@material/web/dialog/dialog';
+import { createComponent } from '@lit-labs/react';
+import * as WordPressElement from '@wordpress/element';
+import DialogTitle from './DialogTitle';
+import DialogContent from './DialogContent';
+import DialogFooter from './DialogFooter';
+
+const MdDialogComponent = createComponent( {
+	tagName: 'md-dialog',
+	elementClass: MdDialog,
+	react: WordPressElement,
+} );
+
+export default function Dialog( { open, children } ) {
+	const slotChildren = WordPressElement.Children.toArray( children ).reduce(
```

Comment:
Note that `Children` is now a legacy API and so its use should be avoided. See https://react.dev/reference/react/Children.